### DISCUSSION TITLE
Custom paragraph

### DISCUSSION THREAD

**Discussion by shadow-light** [ueberdosis/tiptap#2721] [General Questions]
I'm trying to create a custom block that is `<p><small>text...</small></p>` to represent notes (since there is no block level `<small>` in HTML. I couldn't work out how to make `parseHTML` only match paragraphs that have `<small>` as a child, so in addition I set the `.small` class on the paragraph.

But the default paragraph extension overrides this one and it doesn't match at all. If I give it a higher priority than the default paragraph then every paragraph is matched despite `parseHTML` being `p.small`.

Any ideas?

```js
export const Note = Node.create({
    name: 'note',
    group: 'block',
    content: 'inline*',
    defining: true,
    priority: 1001,  // Or less to not override default

    parseHTML(){
        return [{tag: 'p.small'}]
    },

    renderHTML(){
        return ['p', {class: 'small'}, ['small', 0]]
    },
}
```

**Comment by C-Hess** [comment]
You need to set the priority within the parse rule itself (50 is the default, so 51 will ensure it runs first):
```typescript
parseHTML(){
    return [{tag: 'p.small', priority: 51}]
},
```
See Prosemirrors documentation on parse rules here: https://prosemirror.net/docs/ref/#model.ParseRule
